soc/nordic/nrf54h/power.c

```diff
@@ -31,23 +37,43 @@ static void suspend_common(void)
 							RAMBLOCK_CONTROL_BIT_ICACHE, false);
 	}
 
-	/* Disable retention */
-	nrf_lrcconf_retain_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0, false);
-	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0, false);
+	soc_lrcconf_poweron_release(node, NRF_LRCCONF_POWER_DOMAIN_0);
+}
+
+static void common_resume(void)
+{
+	sys_snode_t *node;
+
+	node = soc_pd_sys_snode_get();
+
+	if (IS_ENABLED(CONFIG_ICACHE)) {
+		/* Power up and re-enable ICACHE */
+		nrf_memconf_ramblock_control_enable_set(NRF_MEMCONF, RAMBLOCK_POWER_ID,
+							RAMBLOCK_CONTROL_BIT_ICACHE, true);
+		sys_cache_instr_enable();
+	}
+
+	if (IS_ENABLED(CONFIG_DCACHE)) {
+		/* Power up and re-enable DCACHE */
+		nrf_memconf_ramblock_control_enable_set(NRF_MEMCONF, RAMBLOCK_POWER_ID,
+							RAMBLOCK_CONTROL_BIT_DCACHE, true);
+		sys_cache_data_enable();
+	}
+
+	soc_lrcconf_poweron_request(node, NRF_LRCCONF_POWER_DOMAIN_0);
 }
 
 void nrf_poweroff(void)
 {
 	nrf_resetinfo_resetreas_local_set(NRF_RESETINFO, 0);
 	nrf_resetinfo_restore_valid_set(NRF_RESETINFO, false);
 
+#if !defined(CONFIG_SOC_NRF54H20_CPURAD)
```

Comment:
Should we update these to:
```suggestion
#if !defined(CONFIG_SOC_NRF54H20_CPURAD) && !defined(CONFIG_SOC_NRF54H20_ENGB_CPURAD)
```
or:
```suggestion
#if !defined(NRF_RADIOCORE)
```

Reply:
Done